File: frontend/types/brief.ts
/**
 * Brief 相关类型定义
 * 与后端 BriefResponse 对齐
 */

export interface BriefAttachment {
  id: string
  name: string
  url: string
  size?: string
}

export interface SellingPoint {
  content: string
  priority?: 'core' | 'recommended' | 'reference'
  required?: boolean  // 向后兼容旧格式
}

export interface BlacklistWord {
  word: string
  reason: string
}

// Creative Rubric — AI 自动生成或手动编辑
export interface RubricDimension {
  name: string
  do_items: string[]
  dont_items: string[]
}

export interface CreativeRubric {
  tone?: RubricDimension
  audience?: RubricDimension
  content_style?: RubricDimension
  structure?: RubricDimension
}

export interface BriefResponse {
  id: string
  project_id: string
  project_name?: string | null
  file_url?: string | null
  file_name?: string | null
  product_name?: string | null
  selling_points?: SellingPoint[] | null
  min_selling_points?: number | null
  blacklist_words?: BlacklistWord[] | null
  competitors?: string[] | null
  brand_tone?: string | null
  min_duration?: number | null
  max_duration?: number | null
  other_requirements?: string | null
  attachments?: BriefAttachment[] | null
  agency_attachments?: BriefAttachment[] | null
  creative_rubric?: CreativeRubric | null
  created_at: string
  updated_at: string
}

export interface BriefCreateRequest {
  file_url?: string | null
  file_name?: string | null
  product_name?: string | null
  selling_points?: SellingPoint[]
  min_selling_points?: number | null
  blacklist_words?: BlacklistWord[]
  competitors?: string[]
  brand_tone?: string | null
  min_duration?: number
  max_duration?: number
  other_requirements?: string | null
  attachments?: BriefAttachment[]
  agency_attachments?: BriefAttachment[]
  creative_rubric?: CreativeRubric | null
}

File: backend/app/schemas/brief.py
"""
Brief 相关 Schema

卖点格式 (selling_points: List[dict]):
  新格式: {"content": "卖点内容", "priority": "core|recommended|reference"}
  旧格式: {"content": "卖点内容", "required": true|false}
  兼容规则: required=true → priority="core", required=false → priority="recommended"
"""

from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


# ===== 请求 =====


class BriefCreateRequest(BaseModel):
    """创建/更新 Brief 请求"""

    file_url: Optional[str] = None
    file_name: Optional[str] = None
    product_name: Optional[str] = None
    selling_points: Optional[List[dict]] = None
    min_selling_points: Optional[int] = None
    blacklist_words: Optional[List[dict]] = None
    competitors: Optional[List[str]] = None
    brand_tone: Optional[str] = None
    min_duration: Optional[int] = None
    max_duration: Optional[int] = None
    other_requirements: Optional[str] = None
    attachments: Optional[List[dict]] = None
    agency_attachments: Optional[List[dict]] = None
    creative_rubric: Optional[dict] = None


class BriefUpdateRequest(BaseModel):
    """更新 Brief 请求"""

    file_url: Optional[str] = None
    file_name: Optional[str] = None
    product_name: Optional[str] = None
    selling_points: Optional[List[dict]] = None
    min_selling_points: Optional[int] = None
    blacklist_words: Optional[List[dict]] = None
    competitors: Optional[List[str]] = None
    brand_tone: Optional[str] = None
    min_duration: Optional[int] = None
    max_duration: Optional[int] = None
    other_requirements: Optional[str] = None
    attachments: Optional[List[dict]] = None
    agency_attachments: Optional[List[dict]] = None
    creative_rubric: Optional[dict] = None


class AgencyBriefUpdateRequest(BaseModel):
    """代理商更新 Brief 请求（允许更新代理商附件 + 卖点 + 违禁词 + AI解析内容 + Rubric编辑）"""

    agency_attachments: Optional[List[dict]] = None
    product_name: Optional[str] = None
    selling_points: Optional[List[dict]] = None
    min_selling_points: Optional[int] = None
    blacklist_words: Optional[List[dict]] = None
    brand_tone: Optional[str] = None
    other_requirements: Optional[str] = None
    creative_rubric: Optional[dict] = None


# ===== 响应 =====


class BriefResponse(BaseModel):
    """Brief 响应"""

    id: str
    project_id: str
    project_name: Optional[str] = None
    file_url: Optional[str] = None
    file_name: Optional[str] = None
    product_name: Optional[str] = None
    selling_points: Optional[List[dict]] = None
    min_selling_points: Optional[int] = None
    blacklist_words: Optional[List[dict]] = None
    competitors: Optional[List[str]] = None
    brand_tone: Optional[str] = None
    min_duration: Optional[int] = None
    max_duration: Optional[int] = None
    other_requirements: Optional[str] = None
    attachments: Optional[List[dict]] = None
    agency_attachments: Optional[List[dict]] = None
    creative_rubric: Optional[dict] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
